### PR TITLE
Add unit test

### PR DESCRIPTION
**Pull request by SultanArshad**
Adding unit test in this

### CODE REVIEW

**Review by xxfast — CHANGES_REQUESTED** [review]
Can we also add a test scenario to handle failed states?

**Comment by SultanArshad** [comment]
> Can we also add a test scenario to handle failed states?
sure working on it now

File: app-kotlin/build.gradle [hunk 1]
```diff
@@ -65,11 +66,21 @@ dependencies {
...
   implementation 'io.insert-koin:koin-android:3.2.0'
+
+//  Koin testing tools
+  testImplementation "io.insert-koin:koin-test:3.2.0"
+//  Needed JUnit version
+  testImplementation "io.insert-koin:koin-test-junit4:3.2.0"
+
+//  Mockk Test
+  testImplementation "io.mockk:mockk:1.12.4"
+  androidTestImplementation "io.mockk:mockk-android:1.12.4"
```
Comment (xxfast) [inline app-kotlin/build.gradle:85]: do you need this?

File: app-kotlin/src/test/kotlin/au/com/gridstone/trainingkotlin/DetailDomainTest.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.koin.core.module.Module
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+import org.koin.test.inject
+import retrofit2.Response
+
+class DetailDomainTest : KoinTest {
+  private val viewModel: DetailViewModel by inject()
+  private val service: PokemonService by inject()
+  private val testPokemonId: String = "000000"
```
Comment (xxfast) [inline app-kotlin/src/test/kotlin/au/com/gridstone/trainingkotlin/DetailDomainTest.kt:39]: this can be a `const`

**Review by xxfast — APPROVED** [review]
Nice work 👍

File: app-kotlin/build.gradle (hunk 1 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:85]: duplicated we need one of this of mocking service

**Review by xxfast — APPROVED** [review]

**Pull request merged**